static/app/types/hooks.tsx

```diff
@@ -646,6 +647,20 @@ type InviteButtonCustomizationHook = () => React.ComponentType<{
   organization: Organization;
 }>;
 
+/**
+ * Sidebar navigation item customization allows passing render props to disable
+ * the link, wrap it in an upsell modal, and give it some additional content
+ * (e.g., a Business Icon) to render.
+ */
```

Comment:
maybe let's have a TODO above this that sates how we could replace the label hook with this one potentially